sdk/src/metrics/export/periodic_exporting_metric_reader.cc

```diff
@@ -164,61 +152,39 @@ void PeriodicExportingMetricReader::DoBackgroundWork()
 
 bool PeriodicExportingMetricReader::CollectAndExportOnce()
 {
-  std::atomic<bool> cancel_export_for_timeout{false};
-
   std::uint64_t notify_force_flush = force_flush_pending_sequence_.load(std::memory_order_acquire);
-  std::unique_ptr<std::thread> task_thread;
-
 #if OPENTELEMETRY_HAVE_EXCEPTIONS
   try
   {
 #endif
-    std::promise<void> sender;
-    auto receiver = sender.get_future();
-
-    task_thread.reset(
-        new std::thread([this, &cancel_export_for_timeout, sender = std::move(sender)] {
 #ifdef ENABLE_THREAD_INSTRUMENTATION_PREVIEW
-          if (collect_thread_instrumentation_ != nullptr)
-          {
-            collect_thread_instrumentation_->OnStart();
-            collect_thread_instrumentation_->BeforeLoad();
-          }
+    if (collect_thread_instrumentation_ != nullptr)
+    {
+      collect_thread_instrumentation_->OnStart();
+      collect_thread_instrumentation_->BeforeLoad();
+    }
 #endif /* ENABLE_THREAD_INSTRUMENTATION_PREVIEW */
-
-          this->Collect([this, &cancel_export_for_timeout](ResourceMetrics &metric_data) {
-            if (cancel_export_for_timeout.load(std::memory_order_acquire))
-            {
-              OTEL_INTERNAL_LOG_ERROR(
-                  "[Periodic Exporting Metric Reader] Collect took longer configured time: "
-                  << this->export_timeout_millis_.count() << " ms, and timed out");
-              return false;
-            }
-            this->exporter_->Export(metric_data);
-            return true;
-          });
-
-          const_cast<std::promise<void> &>(sender).set_value();
+    auto start = std::chrono::steady_clock::now();
+    this->Collect([this, &start](ResourceMetrics &metric_data) {
+      auto end = std::chrono::steady_clock::now();
+      if ((end - start) > this->export_timeout_millis_)
+      {
+        OTEL_INTERNAL_LOG_ERROR(
+            "[Periodic Exporting Metric Reader] Collect took longer configured time: "
+            << this->export_timeout_millis_.count() << " ms, and timed out");
+        return false;
+      }
+      this->exporter_->Export(metric_data);
```

Comment:
Note to self - the timeout interval `export_timeout_millis_` is checked at best effort basis between `collect` and `export`. This means, if the collect and/or export takes indefinite time, we can be in blocked state. This is specifically relevant to export method, as its execution time is not in the control of the SDK. The export operation runs to completion once started, regardless of how long it takes.

Nothing for this PR, as this is the existing behavior, but something to discuss further.